packages/twenty-front/src/modules/object-record/record-filter/utils/compute-empty-record-gql-operation-filter/for-composite-field/computeEmptyGqlOperationFilterForEmails.ts

```diff
@@ -0,0 +1,92 @@
+import { FieldMetadataItem } from '@/object-metadata/types/FieldMetadataItem';
+import {
+  EmailsFilter,
+  RecordGqlOperationFilter,
+} from '@/object-record/graphql/types/RecordGqlOperationFilter';
+
+import { RecordFilter } from '@/object-record/record-filter/types/RecordFilter';
+import { isNonEmptyString } from '@sniptt/guards';
+
+export const computeEmptyGqlOperationFilterForEmails = ({
+  recordFilter,
+  correspondingFieldMetadataItem,
+}: {
+  recordFilter: RecordFilter;
+  correspondingFieldMetadataItem: Pick<FieldMetadataItem, 'name' | 'type'>;
+}): RecordGqlOperationFilter => {
+  const subFieldName = recordFilter.subFieldName;
+  const isSubFieldFilter = isNonEmptyString(subFieldName);
+
+  if (isSubFieldFilter) {
+    switch (subFieldName) {
+      case 'primaryEmail': {
+        return {
+          or: [
+            {
+              [correspondingFieldMetadataItem.name]: {
+                primaryEmail: { eq: '' },
+              } satisfies EmailsFilter,
+            },
+            {
+              [correspondingFieldMetadataItem.name]: {
+                primaryEmail: { is: 'NULL' },
+              } satisfies EmailsFilter,
+            },
+          ],
+        };
+      }
+      case 'additionalEmails': {
+        return {
+          or: [
+            {
+              [correspondingFieldMetadataItem.name]: {
+                additionalEmails: { is: 'NULL' },
```

Comment:
and also here we have both [] and NULL, they are actually meaning the same thing. We should consolidate this on BE side

Reply:
Yes, I played a lot with the graphql endpoint to test the different possible values and their results, I agree with you.